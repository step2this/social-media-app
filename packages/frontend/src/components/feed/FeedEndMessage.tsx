import React from 'react';

export const FeedEndMessage: React.FC = () => {
  return (
    <div className="feed-end" data-testid="feed-end" role="status">
      <span className="feed-end__icon" aria-hidden="true">🎉</span>
      <p>You're all caught up!</p>
    </div>
  );
};
