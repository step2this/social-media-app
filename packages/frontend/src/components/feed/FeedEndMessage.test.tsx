import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { FeedEndMessage } from './FeedEndMessage';

describe('FeedEndMessage', () => {
  it('renders end message', () => {
    render(<FeedEndMessage />);
    expect(screen.getByText(/You're all caught up!/i)).toBeInTheDocument();
  });

  it('displays emoji/icon', () => {
    const { container } = render(<FeedEndMessage />);
    const icon = container.querySelector('.feed-end__icon');
    expect(icon).toBeInTheDocument();
  });

  it('shows "You\'re all caught up!" text', () => {
    render(<FeedEndMessage />);
    expect(screen.getByText(/all caught up/i)).toBeInTheDocument();
  });

  it('has correct CSS classes', () => {
    const { container } = render(<FeedEndMessage />);
    const endDiv = container.querySelector('.feed-end');
    expect(endDiv).toBeInTheDocument();
    expect(endDiv).toHaveClass('feed-end');
  });

  it('matches snapshot', () => {
    const { container } = render(<FeedEndMessage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible semantics', () => {
    const { container } = render(<FeedEndMessage />);
    const endDiv = container.querySelector('.feed-end');
    expect(endDiv).toHaveAttribute('role', 'status');
  });

  it('container has test ID', () => {
    render(<FeedEndMessage />);
    expect(screen.getByTestId('feed-end')).toBeInTheDocument();
  });

  it('renders without crashing', () => {
    const { container } = render(<FeedEndMessage />);
    expect(container).toBeInTheDocument();
  });
});
