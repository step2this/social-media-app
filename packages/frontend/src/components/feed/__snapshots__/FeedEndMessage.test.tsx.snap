// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedEndMessage > matches snapshot 1`] = `
<div
  class="feed-end"
  data-testid="feed-end"
  role="status"
>
  <span
    aria-hidden="true"
    class="feed-end__icon"
  >
    ðŸŽ‰
  </span>
  <p>
    You're all caught up!
  </p>
</div>
`;
